enhancements/network/network-policy-audit-logging.md

```diff
@@ -0,0 +1,315 @@
+
+## Motivation
+
+Currently openshift has no ability to audit security events reguarding NetworkPolicies, and there are many customers which require this feature for security policy and regulation. The logging will allow customers to monitor NetworkPolicy events and identify pattterns involving malicious activity with both allow and deny events. This is necessary in scenarios where customers require a certain level of compliance, such as monitoring firewall activity, intrusion detection support, or fo post-mortem analysis 
+
+### Goals
+
+- Activate configurable ACL allow/deny logging in Ovn-Kubernetes on a per namespace basis 
+- Allow the Customer to control the logging rate for this feature via the Cluster Network Operator's config 
+- Allow the Customer to configure this fetaure's use and level via a Namespace's `.metadata.annotations` field 
+- Collect the relevant data from the ovn-controller logs and preset it to the Customer in an seamless manner 
+
+### Non-Goals
+
+- Network Policy Object Logging (`oc describe <entitiy>` takes care of that)
+
+## Proposal
+
+The Enhancment's implementation starts with somce changes to the [network.operator.openshift.io](https://github.com/openshift/api/blob/master/operator/v1/types_network.go) API.  These changes involve adding one optional flag to the `OVNKubernetesConfig` struct as follows. 
+
+```go
+// ovnKubernetesConfig contains the configuration parameters for networks
+// using the ovn-kubernetes network project
+type OVNKubernetesConfig struct {
+	// mtu is the MTU to use for the tunnel interface. This must be 100
+	// bytes smaller than the uplink mtu.
+	// Default is 1400
+	// +kubebuilder:validation:Minimum=0
+	// +optional
+	MTU *uint32 `json:"mtu,omitempty"`
+	// geneve port is the UDP port to be used by geneve encapulation.
+	// Default is 6081
+	// +kubebuilder:validation:Minimum=1
+	// +optional
+	GenevePort *uint32 `json:"genevePort,omitempty"`
+	// HybridOverlayConfig configures an additional overlay network for peers that are
+	// not using OVN.
+	// +optional
+	HybridOverlayConfig *HybridOverlayConfig `json:"hybridOverlayConfig,omitempty"`
+	// ipsecConfig enables and configures IPsec for pods on the pod network within the
+	// cluster.
+	// +optional
+	IPsecConfig *IPsecConfig `json:"ipsecConfig,omitempty"`
+  // aclLoggingRateLimit allows for the configuring of max acl logging rate 
+  // +optional
+  aclLoggingRateLimit  *uint32 `json:"aclLoggingRateLimit,omitempty"`
+
+```
+
+The flag `aclLoggingRateLimit` defaults to `20`. 
```

Comment:
What does`20` imply? 20 logs per second? First 20 logs are written and the rest dropped? Does it tell you when logs are dropped or how many are dropped?

Reply:
20 messages per second, and yeah once that meter argument is surpassed any additional logs are dropped. 

```
 The  action  argument specifies what should happen
 when this meter is exceeded. The only supported action is drop.
```